File: backend/src/services/scraper.rs
pub async fn fetch_html_page(url: &str) -> Result<String, Box<dyn std::error::Error>> {
    let normalized = if url.contains("://") {
        url.to_string()
    } else {
        format!("https://{}", url)
    };

    let parsed = reqwest::Url::parse(&normalized)?;
    let response = reqwest::get(parsed).await?;
    let html = response.text().await?;
    Ok(html)
}

File: backend/services/scraper.go
package services

import (
    "fmt"
    "io"
    "net/http"
    "strings"
)

func FetchHTMLPage(url string) (string, error) {
    normalized := url
    if !strings.Contains(normalized, "://") {
        normalized = "https://" + url
    }

    resp, err := http.Get(normalized)
    if err != nil {
        return "", err
    }
    defer resp.Body.Close()

    if resp.StatusCode != http.StatusOK {
        return "", fmt.Errorf("HTTP error: %d", resp.StatusCode)
    }

    bodyBytes, err := io.ReadAll(resp.Body)
    if err != nil {
        return "", err
    }

    return string(bodyBytes), nil
}